modules/rostests/apitests/iphlpapi/GetInterfaceName.c

```diff
@@ -0,0 +1,459 @@
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        Error = _SEH2_GetExceptionCode();
+        ok(Error == STATUS_SUCCESS, "Exception 0x%08lx, expected 0x%08lx\n", Error, STATUS_SUCCESS);
+    }
+    _SEH2_END;
+    Error = GetLastError();
+
+    ok(ApiReturn == ERROR_SUCCESS, // Windows 7: ERROR_INVALID_PARAMETER
+       "ApiReturn returned %ld, expected ERROR_SUCCESS\n",
+       ApiReturn);
+    ok(Error == ERROR_SUCCESS, // Windows 7: 0xbeeffeed
+       "GetLastError() returned %ld, expected ERROR_SUCCESS\n",
+       Error);
+    ok(ulOutBufLen == 0,
+       "ulOutBufLen is %ld, expected = 0\n",
+       ulOutBufLen);
+    Error = wcslen(Name);
+    ok(Error == 0,
+       "wcslen(Name) is %ld, expected = 0\n",
+       Error);
+    if (Error > 0)
+        trace("Unexpected name: \"%S\"\n", Name);
+
+    // Test correct GUID, but NULL name
+    SetLastError(0xbeeffeed);
+    ZeroMemory(&Name, sizeof(Name));
+    ApiReturn = pTestFunction(&AdapterGUID, NULL, &ulOutBufLen, par1, par2);
+    Error = GetLastError();
+
+    ok(ApiReturn == ERROR_INSUFFICIENT_BUFFER, // Windows 7: ERROR_INVALID_PARAMETER
+       "ApiReturn returned %ld, expected ERROR_INSUFFICIENT_BUFFER\n",
+       ApiReturn);
+    ok(Error == ERROR_SUCCESS,
+       "GetLastError() returned %ld, expected ERROR_SUCCESS\n",
+       Error);
+    ok(ulOutBufLen > 0, // Windows 7: 0
+       "ulOutBufLen is %ld, expected > 0\n",
+       ulOutBufLen);
+    Error = wcslen(Name);
+    ok(Error == 0,
+       "wcslen(Name) is %ld, expected = 0\n",
+       Error);
+    if (Error > 0)
+        trace("Unexpected name: \"%S\"\n", Name);
+
+    // NhGetInterfaceNameFromDeviceGuid will throw exception if pOutBufLen is NULL
+    SetLastError(0xbeeffeed);
+    ZeroMemory(&Name, sizeof(Name));
+    _SEH2_TRY
+    {
+        ApiReturn = pTestFunction(&AdapterGUID, &Name, NULL, par1, par2);
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        Error = _SEH2_GetExceptionCode();
+        ok(Error == STATUS_SUCCESS, "Exception 0x%08lx, expected 0x%08lx\n", Error, STATUS_SUCCESS);
+    }
+    _SEH2_END;
+    Error = GetLastError();
+
+    ok(ApiReturn == ERROR_INSUFFICIENT_BUFFER, // Windows 7: ERROR_INVALID_PARAMETER
+       "ApiReturn returned %ld, expected ERROR_INSUFFICIENT_BUFFER\n",
+       ApiReturn);
+    ok(Error == ERROR_SUCCESS,
+       "GetLastError() returned %ld, expected ERROR_SUCCESS\n",
+       Error);
+    ok(ulOutBufLen > 0, // Windows 7: 0
+       "ulOutBufLen is %ld, expected > 0\n",
+       ulOutBufLen);
+    Error = wcslen(Name);
+    ok(Error == 0,
+       "wcslen(Name) is %ld, expected = 0\n",
+       Error);
+    if (Error > 0)
+        trace("Unexpected name: \"%S\"\n", Name);
+
+    // Test correct values
+    SetLastError(0xbeeffeed);
+    ZeroMemory(&Name, sizeof(Name));
+    ulOutBufLen = sizeof(Name);
+    ApiReturn = pTestFunction(&AdapterGUID, &Name, &ulOutBufLen, par1, par2);
+    Error = GetLastError();
```

Comment:
Not all functions set the error in case of success, keep this in mind.

Reply:
Of course, but this API function is an exception, it sets last error in most cases, even in success.